LadybugTools_Adapter/AdapterActions/Execute.cs

```diff
@@ -399,9 +407,149 @@ private List<object> RunCommand(UTCIHeatPlotCommand command)
+            string result = Engine.Python.Compute.RunCommandStdout(command: cmdCommand, hideWindows: true);
+
+            m_executeSuccess = true;
+            return new List<object>() { result.Split('\n').Last() };
+        }
+
+        /**************************************************/
+
+        private List<object> RunCommand(SunPathPlotCommand command)
+        {
+            if (command.EPWFile == null)
+            {
+                BH.Engine.Base.Compute.RecordError($"{nameof(command.EPWFile)} input cannot be null.");
+                return null;
+            }
+            
+            if (!System.IO.File.Exists(command.EPWFile.GetFullFileName()))
+            {
+                BH.Engine.Base.Compute.RecordError($"File '{command.EPWFile.GetFullFileName()}' does not exist.");
+                return null;
+            }
+
+            if (command.AnalysisPeriod == null)
+            {
+                BH.Engine.Base.Compute.RecordError($"{nameof(command.EPWFile)} input cannot be null.");
+                return null;
+            }
+
+            if (command.SunSize < 1)
+            {
+                BH.Engine.Base.Compute.RecordError($"Sun size must be more than or equal to 1.");
+                return null;
+            }
+
+            string epwFile = System.IO.Path.GetFullPath(command.EPWFile.GetFullFileName());
+
+            string script = Path.Combine(Engine.Python.Query.DirectoryCode(), "LadybugTools_Toolkit\\src\\ladybugtools_toolkit\\bhom\\wrapped\\plot", "sunpath.py");
+
+            //run the process
+            string cmdCommand = $"{m_environment.Executable} {script} -e \"{epwFile}\" -s {command.SunSize} -ap \"{command.AnalysisPeriod.FromBHoM().Replace("\"", "\\\"")}\" -p \"{command.OutputLocation}\"";
+            string result = Engine.Python.Compute.RunCommandStdout(cmdCommand, hideWindows: true);
+
+            m_executeSuccess = true;
+            return new List<object>() { result.Split('\n').Last() };
+        }
+
+        /**************************************************/
+
+        private List<object> RunCommand(SolarPanelTiltOptimisationCommand command)
+        {
+            if (command.EPWFile == null)
+            {
+                BH.Engine.Base.Compute.RecordError($"{nameof(command.EPWFile)} input cannot be null.");
+                return null;
+            }
+
+            if (!System.IO.File.Exists(command.EPWFile.GetFullFileName()))
+            {
+                BH.Engine.Base.Compute.RecordError($"File '{command.EPWFile.GetFullFileName()}' does not exist.");
+                return null;
+            }
+
+            if (command.Azimuths < 3)
+            {
+                BH.Engine.Base.Compute.RecordError($"Azimuths must be more than or equal to 1.");
```

Comment:
The if statement says `3`, which is correct?

Reply:
This is correct, the python method limits to 3 or more